components/brave_wallet/browser/eth_gas_utils.cc

```diff
@@ -0,0 +1,138 @@
+    return false;
+
+  // "pending" is the last element in base_fee_per_gas
+  // "latest" is the 2nd last element in base_fee_per_gas
+  std::string pending_base_fee_per_gas = base_fee_per_gas.back();
+  uint256_t pending_base_fee_per_gas_uint;
+  if (!HexValueToUint256(pending_base_fee_per_gas,
+                         &pending_base_fee_per_gas_uint))
+    return false;
+
+  // We use "double" math and this is unlikely to get hit, so return false
+  // if the value is too big.
+  if (pending_base_fee_per_gas_uint > kMaxSafeIntegerUint64)
+    return false;
+
+  // Compiler crashes without these 2 casts :(
+  double pending_base_fee_per_gas_dbl =
+      static_cast<double>(static_cast<uint64_t>(pending_base_fee_per_gas_uint));
+  // The base fee is not part of the RLP. We only specify priority fee and max
+  // fee. So the excess will be refunded and it will be at most 12.5% It's best
+  // to assume a larger value here so there's a better chance it will get in the
+  // next block and if it's too high it will go back to the user
+  pending_base_fee_per_gas_dbl *= 1.125;
+  pending_base_fee_per_gas_dbl = std::floor(pending_base_fee_per_gas_dbl);
+  // Compiler crashes without these 2 casts :(
+  *suggested_base_fee_per_gas =
+      (uint256_t)(uint64_t)pending_base_fee_per_gas_dbl;
+
+  const uint256_t fallback_priority_fee = 2e9;
+  *low_priority_fee = fallback_priority_fee;
+  *avg_priority_fee = fallback_priority_fee;
+  *high_priority_fee = fallback_priority_fee;
+
+  // Leave it at fallback values if no reward info is passed
+  if (reward.size() == 0)
+    return true;
+
+  std::vector<uint256_t> priority_fee_uints[3];
+  uint256_t* priority_fees[3] = {low_priority_fee, avg_priority_fee,
+                                 high_priority_fee};
+  for (size_t i = 0; i < 3; i++) {
+    uint256_t& current_priority_fee = *(priority_fees[i]);
+    std::vector<uint256_t>& current_priority_fee_uints = priority_fee_uints[i];
+    bool invalid_data = false;
+    // Convert the string priority fees to uints
+    std::transform(reward.begin(), reward.end(),
+                   std::back_inserter(current_priority_fee_uints),
+                   [&](const std::vector<std::string>& v) -> std::size_t {
+                     uint256_t val = fallback_priority_fee;
+                     if (v.size() != 3) {
+                       invalid_data = true;
+                     } else if (!HexValueToUint256(v[i], &val)) {
+                       invalid_data = true;
+                     }
+                     return val;
+                   });
+
+    // We allow no reward info but we don't allow invalid reward info
+    if (invalid_data) {
+      return false;
+    }
+
+    // Sort the priroirty fee uints
+    std::sort(current_priority_fee_uints.begin(),
+              current_priority_fee_uints.end());
+    size_t percentile_index = current_priority_fee_uints.size() * 0.4;
+    // Calculate the avg priorty fee first to be the 40th percentile of the avg
```

Comment:
This comment should be moved to right above line 110, and it's probably also worth mentioning that we are setting it as the initial values for both low and high here too.

Reply:
Fixed in rebase.